src/tools/tests/manifest2proto-test.ts

```diff
@@ -182,6 +193,187 @@ describe('manifest2proto', () => {
+        handle: 'handle2',
+        name: 'c',
+        type: {entity: {schema: {
+          names: ['Z'],
+          fields: {c: {primitive: 'BOOLEAN'}},
+          hash: 'e1ad5776554cda0ee2aad4c45dc0afda2ffc56b9',
+        }}},
+      }
+    ]);
+  });
+
+
+  it('encodes variable particle instance types', async () => {
+    const manifest = await Manifest.parse(`
+      particle Writer1
+        first: writes [Foo {a: Text, c: Text}]
+        second: writes [Bar {a: Text, b: Text}]
+      particle Writer2
+        first: writes [Foo {a: Text, b: Text}]
+        second: writes [Bar {a: Text, c: Text}]
+      particle Reader1
+        first: reads [~f with Foo {a: Text, b: Text, c: Text}]
+        second: reads [{}]
+      particle Reader2
+        first: reads [Foo {}]
+        second: reads [~b with Bar {a: Text, b: Text, c: Text}]
+      
+      recipe
+        h0: create 
+        h2: create 
+        Writer1
+          first: h0
+          second: h1
+        Writer2
+          first: h0
+          second: h1
+        Reader1
+          first: h0
+          second: h1
+        Reader2
+          first: h0
+          second: h1
+    `);
+    const particles = (await toProtoAndBack(manifest)).recipes[0].particles;
+
+    assert.deepStrictEqual(particles, [
+      {
+        specName: 'Writer1',
+        connections: [
+          {
+            name: 'first',
+            handle: 'h0',
+            type: {collection: {collectionType: {entity: {schema: {
+              names: ['Foo'],
+              fields: {a: {primitive: 'TEXT'}, c: {primitive: 'TEXT'}},
+              hash: '08266b26520a746b944249ce1a6f3375e7480178'
+            }}}}}
+          },
+          {
+            name: 'second',
+            handle: 'h1',
+            type: {collection: {collectionType: {entity: {schema: {
+              names: ['Bar'],
+              fields: {a: {primitive: 'TEXT'}, b: {primitive: 'TEXT'}},
+              hash: 'cb189f1044b3f59bf13928fe759f5d72b5913c75'
+            }}}}}
+          },
+        ]
+      },
+      {
+        specName: 'Writer2',
+        connections: [
+          {
+            name: 'first',
+            handle: 'h0',
+            type: {collection: {collectionType: {entity: {schema: {
+              names: ['Foo'],
+              fields: {a: {primitive: 'TEXT'}, b: {primitive: 'TEXT'}},
+              hash: 'c99bd43b5f6e012613062feb5152d073f2e8155d'
+            }}}}}
+          },
+          {
+            name: 'second',
+            handle: 'h1',
+            type: {collection: {collectionType: {entity: {schema: {
+              names: ['Bar'],
+              fields: {a: {primitive: 'TEXT'}, c: {primitive: 'TEXT'}},
+              hash: '68dc41439853353b2238470a84bdc2e9545838d8'
+            }}}}}
+          },
+        ]
+      },
+      {
+        specName: 'Reader1',
+        connections: [
+          {
+            name: 'first',
+            handle: 'h0',
+            type: {collection: {collectionType: {variable: {
```

Comment:
Don't we want the types on handle connection instances (i.e. real handle connections in the recipe, not the handle connections on particle specs) to be concrete types? That's what we decided is right for Plan.kt, and I think having the same decision here makes sense.